docs/modules/ROOT/pages/adr/0040-adr-for-codey-git-ssh-access.adoc

```diff
@@ -0,0 +1,221 @@
+= ADR 0040 - Codey Git SSH Access
+:adr_author:    Marco De Luca
+:adr_owner:     Schedar
+:adr_reviewers: Schedar
+:adr_date:      2025-10-01
+:adr_upd_date:  2025-10-01
+:adr_status:    draft
+:adr_tags:      codey
+
+include::partial$adr-meta.adoc[]
+
+[NOTE]
+.Summary
+====
+This ADR evaluates methods to enable Git over SSH for Codey and compares several architectures to multiplex SSH traffic for hundreds of Codey tenants through a limited number of ingress IPs.
+
+Findings show that cloud provider LoadBalancer products impose hard limits on the number of listeners per IP:
+
+* Cloudscale: up to 100 listeners per LoadBalancer
+* Exoscale: up to 10 listeners per LoadBalancer
+
+These limits make a single-IP for all tenants design infeasible at large scale.
+We must therefore introduce LoadBalancer sharding and evaluate which proxy technology provides the most reliable and maintainable control plane for that model.
+====
+
+== Context
+
+Codey users want a secure way to push/pull Git repositories using SSH keys.
+Because SSH is not SNI-aware, a single IP necessarily implies one unique TCP port per customer instance.
+The design must:
+
+* Work in Kubernetes/OpenShift environments,
+* Support dynamic onboarding/offboarding of tenants,
+* Avoid proxy restarts during updates,
+* Provide high availability,
+* Scale to hundreds of tenants within provider limits.
+
+== Considered Options
+
+=== Option 1: Per-tenant LoadBalancer (distinct IP per instance)
+
+Each Codey customer's Forgejo instance would expose its SSH Service directly as `type: LoadBalancer`, letting the cloud provider allocate a public IP and open TCP/22 for that instance.
+This is the most straightforward design and uses Kubernetes native Service abstraction without any shared proxying layer.
+Network traffic from the Internet terminates directly on each tenant's Forgejo pod, no intermediate routing or multiplexing is required.
+Operationally, this model scales linearly: every new customer gets one LoadBalancer and one public IP.
+
+.Pros:
+* Simple datapath
+* Simpler to operate
+* No lateral movement risk
+
+.Cons:
+* Potentially hundreds of IPs
+* Linear cost
+* Violates consolidation goal
+
+Status: Discarded
```

Comment:
The Context section doesn't do any judgment; this is a neutral space. Judgment comes in the Decision section. See ADR 0001.

Pros / Cons are OK.